components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,183 @@
+
+ To learn more about
+ [UIAlertController](https://developer.apple.com/documentation/uikit/uialertcontroller)
+ or [UIAlertControllerStyleActionSheet](https://developer.apple.com/documentation/uikit/uialertcontrollerstyle/uialertcontrollerstyleactionsheet)
+
+ MDCActionSheetController does not support UIPopoverController, instead it will always be presented
+ in a sheet from the bottom.
+
+ ### Dependencies
+
+ MDCActionSheetController depends on BottomSheet Component.
+ */
+MDC_SUBCLASSING_RESTRICTED
+@interface MDCActionSheetController : UIViewController
+
+/**
+ Designated initializer to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @param message Descriptive text that summarizes a decision in a sentence or two.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title
+                                               message:(nullable NSString *)message NS_DESIGNATED_INITIALIZER;
+
+/**
+ Convenience initializer to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title;
+
+/**
+ Action sheet controllers must be created with actionSheetControllerWithTitle: or
+ with actionSheetControllerWithTitle:message:
+ */
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil NS_UNAVAILABLE;
+
+/**
+ Action sheet controllers must be created with actionSheetControllerwithTitle:
+   or with actionSheetControllerWithTitle:message:
+ */
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_UNAVAILABLE;
+
+/**
+ Adds an action to the action sheet.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ list item in the list of the action sheet.
+ Action buttons will be laid out from top to bottom depending on the order they
+ were added.
+
+ @param action Will be added to the end of MDCActionSheetController.actions.
+ */
+- (void)addAction:(nonnull MDCActionSheetAction *)action;
+
+/**
+ The actions that the user can take in response to the action sheet.
+
+ The order of the actions in the array matches the order in which they were added to the action sheet.
+ */
+@property (nonatomic, nonnull, readonly, copy) NSArray<MDCActionSheetAction *> *actions;
+
+/**
+ The title of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *title;
```

Comment:
If the title and message are nullable why do we have a new designated initializer? 

Either make the title nonnull or allow the other initializers.